viewing_party/party.py

```diff
@@ -1,23 +1,241 @@
+
+    for friend_movie_dict in user_data["friends"]:
+        for movie in friend_movie_dict["watched"]:
+            titles_friends_watched.add(movie["title"])
+    
+    # unique list of movies user's watched but not their friends
+    list_unique_movies = []
+    for movie in user_data["watched"]:
+        if movie["title"] not in titles_friends_watched:
+            list_unique_movies.append(movie)
+            
+    return list_unique_movies
+
+def get_friends_unique_watched(user_data):
+    '''
+    Return a list of dictionaries, that represents a list of movies that:
+    - aleast one of the user's friends have watched, but the user has not watched aka (friends movies) - (user movies)
+    user_data = {
+        'watched': [{...}, {...}, {...}, {...}, {...}, {...}], 
+        friends': [{'watched': [..{},{},{}..], {'watched': [..{},{},{}..]
+        }
+    '''
+
+    list_titles_watched = set()
+    for movie in user_data["watched"]:
+        list_titles_watched.add(movie["title"])
+
+    list_friends_unique_movie = []
+    for friend in user_data["friends"]:
+        for movie_dict in friend["watched"]:
+            if movie_dict["title"] not in list_titles_watched and movie_dict not in list_friends_unique_movie:
+                list_friends_unique_movie.append(movie_dict)
+
+    return list_friends_unique_movie
+
+
+    
+# Wave 4
+def get_available_recs(user_data):
+    '''
+    user_data= {
+        "subscriptions": [ "","","" ],
+        "watched": [ {}, {}, {} ],
+        "friends": [ {"watched": [...{},{}...] }, {"watched": [...{},{}...] }]
+        }
+    '''
+    
+    lod_unique_friends_movie = get_friends_unique_watched(user_data)
+
+    lod_recommended_movies = []
+    for movie in lod_unique_friends_movie:
+        if movie["host"] in user_data["subscriptions"]:
+            lod_recommended_movies.append(movie)
+
+    return lod_recommended_movies
+
+
+# wave 5
+
+def get_new_rec_by_genre(user_data):    
+    '''
+    Return the list of recommended movies....A movie should be added to this list if and only if:
+    -The user has not watched it
+    -At least one of the user's friends has watched
+    -The "genre" of the movie is the same as the user's most frequent genre
+    '''
+    # return lod of movies friends have watched but not user
+    lod_unique_friends_movie = get_friends_unique_watched(user_data)
+
+    # get user's most freq genre
+    user_freq_genre = get_most_watched_genre(user_data)
+
+    lod_recommended_movies = []
+    for movie in lod_unique_friends_movie:
+        if movie["genre"] == user_freq_genre:
+            lod_recommended_movies.append(movie)
+
+    return lod_recommended_movies
+
+def get_rec_from_favorites(user_data):
+    '''
+    user_data= {
+        "favorites": [ {}, {}, {} ],
+        "watched": [ {}, {}, {} ],
+        "friends": [ {"watched": [...{},{}...] }, {"watched": [...{},{}...] }]
+        }
+    '''
+    # list of movies only the user has seen (not any of the friends)

```

Comment:
Since you're using docstrings throughout the project, I'd prefer this function's docstring to have more details about what the function returns instead of having this detail as an inline comment in the function.